### PR TITLE
TEN-1380 / 24.10.0 / SSH Password Login Groups Access

### PR DESCRIPTION
**Pull request by jordan-mcgillivray005**

### CODE REVIEW

**Comment by jordan-mcgillivray005** [comment]
image

File: keywords/webui/ssh_connection.py [hunk 1]
```diff
@@ -32,6 +34,21 @@ def assert_shell_type(cls, shell_return: str, shell_type: str, username: str = p
...
+    @classmethod
+    def assert_ssh_access(cls, username: str = private_config['USERNAME'], password: str = private_config['PASSWORD']) -> bool:
+        """
+        This method verifies the given user is able to access the system via SSH connection.
+
+        :param username: is the username of the user
+        :param password: is the password of the user
+
+        Example:
+            - SSH_Connection.assert_shell_type('admin', 'bash')
```
Comment (markbainix) [inline keywords/webui/ssh_connection.py:46]: these need fixing. wrong method name and parameters

File: keywords/webui/ssh_connection.py [hunk 2]
```diff
@@ -265,6 +282,30 @@ def set_admin_credentials(cls, user: str, password: str) -> None:
         COM.set_input_field('admin-username', user)
         COM.set_input_field('password', password)
 
+    @classmethod
+    def set_just_input_field(cls, value: str, tab: bool = True, pill: bool = True) -> None:
```
Comment (markbainix) [inline keywords/webui/ssh_connection.py:286]: Suggest: this code is very similar to `set_input_field()` could we combine them or create a 3rd method that these two call?  Reduce duplicate code.

File: test_cases/webui/ssh/test_ssh_controlled_access.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+@allure.feature("SSH Connections")
+class Test_SSH_Controlled_Access:
+
+    @pytest.fixture(scope='class', autouse=True)
+    def setup_class(self, ssh_controlled_access) -> None:
+        """
+        This method sets up the class by starting the SSH service and creating the needed users and groups.
+        """
+        API_POST.start_service('ssh')
+        API_POST.create_group(ssh_controlled_access['group-name'])
```
Comment (markbainix) [inline test_cases/webui/ssh/test_ssh_controlled_access.py:29]: Is is worth having a datafile for just 2 data points?  will this increase in the future? Couldn't we just use 2 strings for this instead?

File: xpaths/common_xpaths.py [hunk 4]
```diff
@@ -43,16 +43,20 @@ def any_header(text: str, level: int) -> str:
...
     This function sets the xpath for the given pill
 
     :param name: name of the input associated with the pill
     :param text: text of the pill
+    :param datatest_just_input: True if the pill's data-test is just 'input'
     :return: xpath string for given pill
     """
-    # return f'//*[@data-test="input-{name}"]/preceding-sibling::mat-chip-row/descendant::*[contains(text(),"{text}")]'
-    return f'//*[@data-test="input-{name}"]/ancestor::mat-chip-grid/descendant::*[contains(text(),"{text}")]'
+    if datatest_just_input:
```
Comment (markbainix) [inline xpaths/common_xpaths.py:55]: Suggestion: could this be simplified to 
`name = f'-{name}' if name`
`return f'//*[@data-test="input{name}"]/...`
that way in the "just input" pass in name = '' and avoid the bool

**Review by markbainix — CHANGES_REQUESTED** [review]

File: xpaths/common_xpaths.py (hunk 4 above)
Comment (jordan-mcgillivray005) [reply xpaths/common_xpaths.py]: Done.

File: test_cases/webui/ssh/test_ssh_controlled_access.py (hunk 3 above)
Comment (jordan-mcgillivray005) [reply test_cases/webui/ssh/test_ssh_controlled_access.py]: I agree. Removed it.

File: keywords/webui/ssh_connection.py (hunk 2 above)
Comment (jordan-mcgillivray005) [reply keywords/webui/ssh_connection.py]: Done via if name: name = "-" + name.

File: keywords/webui/ssh_connection.py (hunk 1 above)
Comment (jordan-mcgillivray005) [reply keywords/webui/ssh_connection.py]: Fixed.

**Review by markbainix — APPROVED** [review]

**Pull request merged**